core/config/project_settings.cpp

```diff
@@ -1256,6 +1256,12 @@ ProjectSettings::ProjectSettings() {
 
 	GLOBAL_DEF("compression/formats/gzip/compression_level", Compression::gzip_level);
 	custom_prop_info["compression/formats/gzip/compression_level"] = PropertyInfo(Variant::INT, "compression/formats/gzip/compression_level", PROPERTY_HINT_RANGE, "-1,9,1");
+
+	// Required to make the project setting appear even if the physics engine is GodotPhysics,
+	// while also making it appear in the ProjectSettings class documentation.
+	// The setting is only relevant when using Bullet physics, but it should still
+	// appear in the project settings as documented.
+	GLOBAL_DEF("physics/3d/smooth_trimesh_collision", false);
```

Comment:
There are a couple other `physics/` settings above, would be nice to keep them grouped. Maybe the comment can also be shortened a bit to reduce LOC / information overload, two lines would be enough to convey why this is here IMO.